pip/commands/install.py

```diff
@@ -261,7 +266,22 @@ def run(self, options, args):
                     return
 
                 try:
-                    requirement_set.prepare_files(finder)
+                    if options.download_dir or not wheel:
+                        # on -d don't do complex things like building
+                        # wheels, and don't try to build wheels when wheel is
+                        # not installed.
+                        requirement_set.prepare_files(finder)
+                    else:
+                        # build wheels before install.
+                        wb = WheelBuilder(
+                            requirement_set,
+                            finder,
+                            build_options=[],
+                            global_options=[],
```

Comment:
Wondering if this is not an issue (for `build_options` and `global_options`)... maybe the use of these would disable the wheel caching as I dont think we want  to build wheels for all possible options ?